apps/costcontrol/test/unit/cost_control_test.js

```diff
@@ -498,4 +498,60 @@ suite('Cost Control Service Hub Suite >', function() {
       });
     }
   );
+
+    test(
```

Comment:
With the second cache approach, the test should become simpler as you only need to check the proper cache is filled with earch query type and the other remain the same.